torchmetrics/functional/text/rouge.py

```diff
@@ -303,14 +333,20 @@ def rouge_score(
             raise ValueError(f"Got unknown rouge key {key}. Expected to be one of {list(ALLOWED_ROUGE_KEYS.keys())}")
     rouge_keys_values = [ALLOWED_ROUGE_KEYS[key] for key in rouge_keys]
 
+    if isinstance(targets, list) and bool(targets) and all(isinstance(elem, str) for elem in targets):
+        if isinstance(preds, str):
+            targets = [str(x) for x in targets]
+        else:
+            targets = [[str(x)] for x in targets]
+
```

Comment:
```suggestion
    if isinstance(targets, list) and all(isinstance(target, str) for target in targets) and isinstance(preds, list):
        targets = [[target] for target in targets]

```

We shouldn't be required to call `str(target)` as this condition is already verified by the preceding condition.

Reply:
Thanks for the suggestions. I will incorporate those.

For this particular case:
1. mypy was raising some errors owing to which I typecasted it.
2. There might be a small issue with the previous if-else. This is the new one I am trying:
```
    if isinstance(targets, list) and bool(targets) and all(isinstance(elem, str) for elem in targets):
	        if isinstance(preds, str):
	            targets = [[str(x) for x in targets]]
	        else:
	            targets = [[str(x)] for x in targets]
```
For handling cases where 

Input
pred = 'this' 
target = ['this', 'that']

Output 
pred = ['this']
target = [['this', 'that']]

Let me know your thoughts